src/sentry/integrations/utils/code_mapping.py

```diff
@@ -505,24 +523,27 @@ def find_roots(stack_path: str, source_path: str) -> tuple[str, str]:
     stack_path_delim = SLASH if SLASH in stack_path else BACKSLASH
     if stack_path_delim == BACKSLASH:
         stack_path = stack_path.replace(BACKSLASH, SLASH)
+    if (straight_path_idx := get_straight_path_prefix_end_index(stack_path)) > 0:
+        stack_root += stack_path[:straight_path_idx]
+        stack_path = stack_path[straight_path_idx:]
+
+    overlap_to_check: list[str] = stack_path.split(SLASH)
+    stack_root_items: list[str] = []
 
-    idx = 0
-    while idx < len(stack_path):
-        if source_path.endswith(overlap := stack_path[idx:]):
+    while overlap_to_check:
+        if (overlap := SLASH.join(overlap_to_check)) and source_path.endswith(overlap):
             source_root = source_path.rpartition(overlap)[0]
+            stack_root += stack_path_delim.join(stack_root_items)
 
-            if stack_root:  # append trailing slash
+            if stack_root and stack_root[-1] != SLASH:  # append trailing slash
```

Comment:
In certain circumstances, we would append a trailing slash when there already existed one, this prevents that from happening.